connect/src/client/hb.js

```diff
@@ -72,14 +72,36 @@ export function requestWith ({ fetch, logger: _logger, HB_URL, signer }) {
         toHttpSigner(signer)(toSigBaseArgs({
           url: joinUrl({ url: HB_URL, path }),
           method,
-          headers,
-          includePath: true
+          headers
+          // includePath: true
         })).then((req) => ({ ...req, body }))
       ))
       .map(logger.tap('Sending HTTP signed message to HB: %o'))
       .chain((request) => of(request)
-        .chain(fromPromise(({ url, method, headers, body }) =>
-          fetch(url, { method, headers, body, redirect: 'follow' })
+        .chain(fromPromise(({ url, method, headers, body }) => {
+          return fetch(url, { method, headers, body, redirect: 'follow' })
+            .then(async res => {
+              if (res.status < 300) {
```

Comment:
Nothing is returned if the `res.status` is greater than 300. Should probably be a clause to just return `res`

Reply:
I don't follow everything else returns res

Reply:
For example, `res.status` is 400, it looks like the `if` check won't match, and then the function ends, which means it returns `undefined`. Shouldn't there be like `else { return res }` or something of the sort?